docs/DEVELOPING.md

```diff
@@ -717,6 +717,188 @@ $ make ftest NAME=mssql
+  - Sharepoint 2019
+
+
+### Create an Administrator Account For SharePoint Online:
+  - To create an administrator account, join the Microsoft 365 Developer Program and get the Microsoft 365 developer subscription with a new administrator account for 90 days. Use this administrator account for signing into SharePoint Online and Azure AD to fetch more details.
+  - Register your app with Microsoft Azure AD
+  - To use the Microsoft identity platform endpoint, you must register your app using the Azure app registration portal. You can use either a Microsoft account or a work or school account to register an app.
+  - To configure an app to use the OAuth 2.0 authorization code grant flow, you'll need to save the following values when registering the app:
+    - Client Id: `Application(client) ID` which you can find inside App registrations from azure portal
+    - Client Secret: It is a `Value` which you need to create  form `Certificates and secrets` from azure portal
+    - Tenant Id: Dedicated instance of azure AD  which you can find inside App registrations from azure portal
+    - Tenant name: Name of domain
+
+
+### SharePoint Online permissions required to run the connector:
+
+   Refer to the following documentation for setting [SharePoint Permissions](https://learn.microsoft.com/en-us/sharepoint/dev/solution-guidance/security-apponly-azureacs)
+
+   - To set `DisableCustomAppAuthentication` to false we need to connect to SharePoint using Windows PowerShell and then run set-spotenant -DisableCustomAppAuthentication $false
+
+   - In azure portal > Azure Active Directory > App registrations > API permission
+        - Click Add a permission
+        - Click Microsoft Graph and complete the following steps:
+            - Click Application permissions.
+            - Select the User.Read permission.
+        - After adding permission Grant admin consent for tenant_name    
+   - To assign full permissions to the tenant in SharePoint Online, in your browser, go to the tenant URL. For example, go to `https://<office_365_tenant_URL>/_layouts/15/appinv.aspx`. The SharePoint admin center page appears.
+       - In the App ID box, enter the application ID that you recorded earlier, and then click Lookup.In the Title box, the name of the application appears.
+       - In the App Domain box, type tenantname.onmicrosoft.com
+       - In the App's Permission Request XML box, type the following XML string:
+           ```
+            <AppPermissionRequests AllowAppOnlyPolicy="true">
+              <AppPermissionRequest Scope="http://sharepoint/content/tenant" Right="FullControl" />
+              <AppPermissionRequest Scope="http://sharepoint/social/tenant" Right="Read" />
+            </AppPermissionRequests>
+            ```
+
+### Objects to be indexed
+  - Sites and Subsites
+  - Lists
+  - List Items and its attachment content
+  - Document Libraries and its attachment content(include Web Pages HTML content)
+
+### Setup and basic usage
```

Comment:
@leemthompo suggested to use the official structure as discussed in [slack](https://elastic.slack.com/archives/C02AV5C8ZDE/p1680252430113829)

Should we maybe use "Configuration"  for this step?

Reply:
I'd use the **Usage** heading here for this info. :)

Reply:
Refer to the pattern in[ S3 connector docs](https://www.elastic.co/guide/en/enterprise-search/current/connectors-s3.html#connectors-s3-usage)

Reply:
@akanshi-elastic could you please address @leemthompo comment?